### PR TITLE
Finished the filter component and added a get all categories function in the category service.

### PR DESCRIPTION
**Pull request by talosdennis**

### CODE REVIEW

File: src/app/components/generic-filter/generic-filter.component.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+	categoriesPool: string[];
+
+	sortMethods: string[] = ['Start Date', 'Join Deadline', 'Title', 'Fee'];
+
+	constructor(private categoryService: CategoryService) {}
+
+	ngOnInit(): void {
+		this.categoryService.getCategories().subscribe((categories) => {
+			this.categoriesPool = categories.result.map((el) => el.title);
```
Comment (larisaChh) [inline src/app/components/generic-filter/generic-filter.component.ts]: Here you also need the ids. To the be we need to send the ids, not the titles.
You can let categoriesPool be the actual response and in the template map it like this:
<mat-option *ngFor="let category of categoriesPool" [value]="category.id">{{ category.title}}</mat-option>

File: src/app/components/generic-filter/generic-filter.component.html [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+					</mat-form-field>
+			</div>
+		</div>
+		<div class="multiple-select">
+			<label> Category </label>
+			<mat-form-field appearance="fill" class="filter-select-input">
+				<mat-label>Choose categories</mat-label>
+				<mat-select
+					multiple
+					[(value)]="filtersDataSelected.categories"
```
Comment (larisaChh) [inline src/app/components/generic-filter/generic-filter.component.html:111]: The value here should be  a list of ids

File: src/app/models/filters-data.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export interface FiltersData {
+    startDate:Date,
+    endDate?:Date,
+    categories?:string[],
```
Comment (larisaChh) [inline src/app/models/filters-data.ts]: This should be a list of ids

File: src/app/components/generic-filter/generic-filter.component.ts [hunk 4]
```diff
@@ -45,6 +46,7 @@ export class GenericFilterComponent implements OnInit {
 
 	emitFiltersData() {
 		this.filtersData.emit(this.filtersDataSelected);
+		console.log(this.filtersDataSelected);
```
Comment (indreiuflavius) [inline src/app/components/generic-filter/generic-filter.component.ts]: remove this

**Pull request merged**